src/titiler/core/titiler/core/templates/index.html

```diff
@@ -69,79 +36,63 @@
     }
 }
 
-const addAOI = (bounds) => {
-  if (map.getLayer('aoi-polygon')) map.removeLayer('aoi-polygon')
-  if (map.getSource('aoi')) map.removeSource('aoi')
-  const geojson = {
-      "type": "FeatureCollection",
-      "features": [bboxPolygon(bounds)]
+// tms comes from morecantile
+// https://github.com/developmentseed/morecantile/blob/main/morecantile/models.py
+var crs = new L.Proj.CRS(
+    '{{ tms.crs.srs }}',
+    '{{ tms.crs.to_proj4() }}', {
+    origin: [{{ tms.xy_bbox[0] }}, {{ tms.xy_bbox[3] }}],
+    // bounds: [[{{ tms.xy_bbox[0] }}, {{ tms.xy_bbox[1] }}], [{{ tms.xy_bbox[2] }}, {{ tms.xy_bbox[3] }}]],
+    resolutions: {{ resolutions|safe }},
   }
+);
 
-  map.addSource('aoi', {
-    'type': 'geojson',
-    'data': geojson
-  })
 
-  map.addLayer({
-    id: 'aoi-polygon',
-    type: 'line',
-    source: 'aoi',
-    layout: {
-      'line-cap': 'round',
-      'line-join': 'round'
-    },
-    paint: {
-      'line-color': '#3bb2d0',
-      'line-width': 1
-    }
-  })
-  return
-}
+var map = L.map('map', {
+  zoom: 0,
+  center: [0, 0],
+  crs: crs,
+  continuousWorld: true,
+  worldCopyJump: false
+});
 
-map.on('load', () => {
+fetch('{{ tilejson_endpoint|safe }}')
+  .then(res => {
+    if (res.ok) return res.json()
+    throw new Error('Network response was not ok.')
+  })
+  .then(data => {
+    console.log(data)
 
-  fetch('{{ tilejson_endpoint|safe }}')
-    .then(res => {
-      if (res.ok) return res.json()
-      throw new Error('Network response was not ok.')
-    })
-    .then(data => {
-      console.log(data)
+    let bounds = [...data.bounds]
+    // Bounds crossing dateline
+    if (bounds[0] > bounds[2]) {
+      bounds[0] = bounds[0] - 360
+    }
+    var left = bounds[0],
+      bottom = bounds[1],
+      right = bounds[2],
+      top = bounds[3];
 
-      let bounds = [...data.bounds]
-      // Bounds crossing dateline
-      if (bounds[0] > bounds[2]) {
-        bounds[0] = bounds[0] - 360
+    var aoi = L.geoJSON(
+      bboxPolygon(bounds), {
+        color: '#3bb2d0', fill: false
       }
-      map.fitBounds(
-        [[bounds[0], bounds[1]], [bounds[2], bounds[3]]]
-      )
-      addAOI(bounds)
-
-      map.addSource(
-        'raster',
-        {
-          type: 'raster',
-          bounds: data.bounds,
-          minzoom: data.minzoom,
-          maxzoom: data.maxzoom,
-          tiles: data.tiles,
-        }
-      )
+    ).addTo(map);
+    map.fitBounds(aoi.getBounds());
 
-      map.addLayer(
-        {
-          id: 'raster',
-          type: 'raster',
-          source: 'raster',
-        }
-      )
-
-    })
-    .catch(err => {
-      console.warn(err)
-    })
-})
+    L.tileLayer(
+      data.tiles[0], {
+        minZoom: data.minzoom,
+        maxZoom: data.maxzoom,
+        bounds: L.latLngBounds([bottom, left], [top, right]),
```

Comment:
I don't think you want to do this - the `bounds` option on a `L.Tilelayer` is a **constraint**, so the tilelayer will load less tiles. It's not an informative field.

Focus on getting the `bounds` of the Leaflet `CRS` right - tilelayers will work throughout that range, and *only* throughout that range.